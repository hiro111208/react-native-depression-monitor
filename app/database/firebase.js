//import firebase from "firebase/app";
import "firebase/firestore";
import firebase from 'firebase'
require('firebase/auth')

const firebaseConfig = {
  apiKey: "AIzaSyDwOUkdxC7kNNILoFTSx05PTUxRci_kcaw",
  authDomain: "fireship-tutorial-646fc.firebaseapp.com",
  projectId: "fireship-tutorial-646fc",
  storageBucket: "fireship-tutorial-646fc.appspot.com",
  messagingSenderId: "289064122528",
  appId: "1:289064122528:web:ee39abfa2183c8499071c5",
  measurementId: "G-D35RDH9V0E",
};

firebase.initializeApp(firebaseConfig);

export default firebase;
