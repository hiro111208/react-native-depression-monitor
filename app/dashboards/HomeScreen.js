import React, { useState, useEffect } from "react";
import {
  StyleSheet,
  View,
  Text,
  Button,
  Image,
  TouchableOpacity,
} from "react-native";
import firebase from "../database/firebase";
import * as indexStyles from "../config/indexStyles";
import colors from "../config/colors";
import { Alert, Platform } from "react-native";
import * as Permissions from "expo-permissions";
import { Linking } from "expo";
import * as Notifications from "expo-notifications";
//import firebase from firebase;
import Constants from "expo-constants";

export default function HomeScreen({ route, props, navigation }) {
  const [user, setUser] = useState(undefined);
  const [plant, setPlant] = useState(require("../assets/stage_1.png"));
  const [displayName, setDisplayName] = useState(
    firebase.auth().currentUser !== null
      ? firebase.auth().currentUser.displayName
      : ""
  );

  useEffect(() => {
    getLevel();
  }, []);

  function getLevel() {
    firebase
      .firestore()
      .collection("users")
      .doc(firebase.auth().currentUser.uid)
      .get()
      .then((doc) => {
        setUser(doc.data());
        const level = doc.data().level;
        updatePath(level);
      })
      .catch((error) => {
        console.log("Error getting document:", error);
      });
  }

  function updatePath(lvl) {
    switch (lvl) {
      case 1:
        setPlant(require("../assets/stage_1.png"));
        break;
      case 2:
        setPlant(require("../assets/stage_2.png"));
        break;
      case 3:
        setPlant(require("../assets/stage_3.png"));
        break;
      case 4:
        setPlant(require("../assets/stage_4.png"));
        break;
      case 5:
        setPlant(require("../assets/stage_5.png"));
        break;
      case 6:
        setPlant(require("../assets/stage_6.png"));
        break;
      case 7:
        setPlant(require("../assets/stage_7.png"));
        break;
      case 8:
        setPlant(require("../assets/stage_8.png"));
        break;
      case 9:
        setPlant(require("../assets/stage_9.png"));
        break;
      default:
        setPlant(require("../assets/stage_9.png"));
    }
  }

  function refresh() {
    getLevel();
  }

  ///

  useEffect(() => {
    (async () => {
      const user = await firebase
        .firestore()
        .collection("users")
        .doc(firebase.auth().currentUser.uid)
        .get();
    })();
  });

  useEffect(() => {
    (() => registerForPushNotificationsAsync())();
  }, []);

  const registerForPushNotificationsAsync = async () => {
    let token; // undefined initially
    if (Constants.isDevice) {
      const { status: existingStatus } = await Permissions.getAsync(
        Permissions.NOTIFICATIONS
      );
      let finalStatus = existingStatus;
      if (existingStatus !== "granted") {
        const { status } = await Permissions.askAsync(
          Permissions.NOTIFICATIONS
        );
        finalStatus = status;
      }
      if (finalStatus !== "granted") {
        return false;
      }
      token = (await Notifications.getExpoPushTokenAsync()).data;
      console.log(token);
    } else {
      alert("Must use physical device for Push Notifications");
    }
    // save the users token in firebase
    if (token) {
      //makes sure there is a token to add to firebase
      const res = await firebase
        .firestore()
        .collection("users")
        .doc(firebase.auth().currentUser.uid)
        .set({ token }, { merge: true });
    }

    if (Platform.OS === "android") {
      Notifications.setNotificationChannelAsync("default", {
        name: "default",
        importance: Notifications.AndroidImportance.MAX,
        vibrationPattern: [0, 250, 250, 250],
        lightColor: "#FF231F7C",
      });
    }

    return token;
  };

  ///

  return (
    <View style={styles.container}>
      <View style={styles.center}>
        <View style={[styles.welcomeArea, indexStyles.shadowEffect]}>
          <View style={styles.userNote}>
            <Text style={[indexStyles.textGrey]}>Hello there, {displayName}!</Text>

            <View style={styles.spacer}></View>

            <View
              style={[
                styles.plantImage,
                indexStyles.centering,
                indexStyles.shadowEffect,
              ]}
            >
              <Image
                style={{ width: "100%", height: "100%" }}
                resizeMode="contain"
                source={plant}
              />
            </View>
          </View>

          <View style={{ height: "30%" }}></View>

          <TouchableOpacity
            style={[
              styles.sessionArea,
              indexStyles.centering,
              indexStyles.shadowEffect,
            ]}
            onPress={() => {
              if (user.question === 1) {
                navigation.navigate("LogFeelingScreen", {
                  cameFrom: "HomeScreen",
                  onGoBack: () => refresh(),
                });
              } else {
                navigation.navigate("TherapyScreen", {
                  onGoBack: () => refresh(),
                });
              }
            }}
          >
            <Text style={indexStyles.textGrey}>Go to your session</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={[
              styles.sessionArea,
              indexStyles.centering,
              indexStyles.shadowEffect,
            ]}
            onPress={() =>
              navigation.navigate("PlantScreen", {
                currentUser: user,
                onGoBack: () => refresh(),
              })
            }
          >
            <Text style={indexStyles.textGrey}>Interact with your plant</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    padding: 25,
    backgroundColor: "#fff",
  },
  center: {
    height: "100%",
    width: "100%",
    alignItems: "center",
  },
  welcomeArea: {
    width: "100%",
    height: "100%",
    borderRadius: 50,
    backgroundColor: "#fed8b1",
    alignItems: "center",
  },
  userNote: {
    height: "30%",
    width: "100%",
    backgroundColor: "#ffeed2",
    alignItems: "center",
    borderTopStartRadius: 50,
    borderTopEndRadius: 50,
    padding: 20,
  },
  sessionArea: {
    height: "17%",
    width: "90%",
    backgroundColor: "#ffeed2",
    padding: 10,
    borderRadius: 20,
  },
  spacer: {
    height: "20%",
  },
  plantImage: {
    width: 225,
    height: 225,
    borderRadius: 100,
    borderWidth: 4,
    borderColor: "#fff",
    backgroundColor: "#eee",
  },
});
