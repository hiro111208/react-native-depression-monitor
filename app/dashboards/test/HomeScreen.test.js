import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import HomeScreen from "../HomeScreen";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import * as firebase from "firebase";

beforeEach(() => {
  jest.useFakeTimers();
});

describe("Testing HomeScreen.js rendering", () => {

  jest.spyOn(firebase, "auth").mockImplementation(() => {
    return {
      currentUser: {
        displayName: "TestUser",
        uid: "123456"
      },
    }
  });

  it("HomeScreen renders correctly", () => {
    const wrapper = shallow(<HomeScreen />);
    expect(wrapper.length).toBe(1);
  });
  
  it("HomeScreen snapshot test", () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders plant image", () => {
    const wrapper = render(<HomeScreen />);
    expect(wrapper.queryByTestId("PLANT_IMAGE")).toBeTruthy();
  });

  it("Renders session button", () => {
    const wrapper = render(<HomeScreen />);
    expect(wrapper.queryByTestId("SESSION_BUTTON")).toBeTruthy();
  });

  it("Renders plant button", () => {
    const wrapper = render(<HomeScreen />);
    expect(wrapper.queryByTestId("PLANT_BUTTON")).toBeTruthy();
  });

});

describe("Testing HomeScreen.js navigation", () => {

  //TODO: PLANT_BUTTON navigates to PlantScreen
  it("PLANT_BUTTON navigates to LogFeelingScreen when user question is 0", () => {
    const navigation = { navigate: jest.fn() };
    const wrapper = render(<HomeScreen navigation={navigation}/>)
    const plantButton = wrapper.getByTestId("PLANT_BUTTON");
    expect(navigation.navigate).not.toHaveBeenCalledWith("PlantScreen");
    fireEvent(plantButton, "press");
    expect(navigation.navigate).toHaveBeenCalled();
  });

});

afterAll(cleanup);