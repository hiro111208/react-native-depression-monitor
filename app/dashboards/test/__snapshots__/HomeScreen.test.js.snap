// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen.js rendering HomeScreen snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "display": "flex",
        "flex": 1,
        "padding": 25,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fed8b1",
            "borderColor": "#ffeed2",
            "borderRadius": 50,
            "borderWidth": 5,
          },
          Object {
            "elevation": 5,
            "marginVertical": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "borderTopEndRadius": 40,
            "borderTopStartRadius": 40,
            "height": "30%",
            "padding": 20,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "dimgray",
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Hello there, 
          TestUser
          !
        </Text>
        <View
          style={
            Object {
              "height": "20%",
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#eee",
                "borderColor": "#fff",
                "borderRadius": 100,
                "borderWidth": 4,
                "height": 225,
                "width": 225,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "elevation": 5,
                "marginVertical": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              },
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../app/assets/stage_1.png",
              }
            }
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
            testID="PLANT_IMAGE"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "30%",
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "borderRadius": 20,
            "elevation": 5,
            "height": "17%",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
        testID="SESSION_BUTTON"
      >
        <Text
          style={
            Object {
              "color": "dimgray",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Go to your session
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "borderRadius": 20,
            "elevation": 5,
            "height": "17%",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "90%",
          }
        }
        testID="PLANT_BUTTON"
      >
        <Text
          style={
            Object {
              "color": "dimgray",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Interact with your plant
        </Text>
      </View>
    </View>
  </View>
</View>
`;
